l2r_part1_data_eng.ipynb

```diff
@@ -242,8 +310,8 @@
     "# filter out queries with only one document. These are trivial and useless.\n",
     "query_groups = judgments.groupby('query').filter(lambda x: len(x) > 1).groupby('query')\n",
     "\n",
-    "executor = concurrent.futures.ThreadPoolExecutor(30)\n",
-    "futures = [executor.submit(get_features_from_service, azs_service, query, group) for (query, group) in query_groups]\n",
+    "executor = concurrent.futures.ThreadPoolExecutor(15)\n",
```

Comment:
Any reason you dropped it down to 15 from 30? It is likely a better choice considering these docs I stumbled upon: https://docs.python.org/3/library/concurrent.futures.html